### PR TITLE
More information on the parse function.

### PR DESCRIPTION
**Pull request by udohjeremiah**
Providing more information on how the parse function works with floating point literals would be great, as I've ran into situations where I thought it intuitive to work with the `Float32` numeric literal, but it throws an error.

Providing a `!!! note` up-front about this behaviour and how to go around it would be great to inform users to take notice of such occurrences.

### CODE REVIEW

**Comment by udohjeremiah** [comment]
I've had a rethink on this and it doesn't seems "elegant" to provide a `!!! note` on a case where there's just one exception.

So its best to include the exception on `# Example` and then show how to bypass it - in the example as well.

This seems concise and elegant and would even give users **"an hint"** to try out other numeric literals and see if they work as well - which they all do (to my best knowledge). And even if they don't, that can lead to a potential useful PR or even an ISSUE.

**Comment by udohjeremiah** [comment]
@KristofferC is still being looked into?

File: base/parse.jl [hunk 1]
```diff
@@ -31,6 +31,12 @@ julia> parse(Int, "afc", base = 16)
 julia> parse(Float64, "1.2e-3")
 0.0012
 
+julia> parse(Float32, "24.9f-3")     # f literal throws an error
+ERROR: ArgumentError: cannot parse "24.9f-3" as Float32
+
+julia> parse(Float32, "$(24.9f-3)")  # interpolate to bypass it
+0.0249f0
+
```
Comment (mcabbott) [inline base/parse.jl:39]: This just parses the string "0.0249". But seems unlikely to be the solution to anyone's problem, why parse if you have the number?
```suggestion
julia> parse(Float32, "$(24.9f-3)")  # interpolate to bypass it
0.0249f0

```

Comment (KristofferC) [reply base/parse.jl:39]: Yeah, interpolating a number into a string to then parse it doesn't make much sense.

Comment (mcabbott) [reply base/parse.jl:39]: Should this be removed before merging? I see you added the label.

Comment (KristofferC) [reply base/parse.jl:39]: Must have added it to the wrong PR.

**Comment by KristofferC** [comment]
We could add more details of the type of formats we accept in `parse`. For example:

```
julia> parse(Float64, "0x2.3ep5")
71.75

julia> parse(Int, "0o777")
511
```

but just adding a case of Julia syntax that doesn't parse doesn't seem worthwhile.

**Pull request closed without merging**